glib/src/value.rs

```diff
@@ -271,6 +271,10 @@ impl Value {
     ) -> Result<SendValue, Self> {
         self.downcast::<T>().map(TypedValue::into_send_value)
     }
+
+    fn content_debug_string(&self) -> GString {
+        unsafe { from_glib_full(gobject_ffi::g_strdup_value_contents(self.to_glib_none().0)) }
```

Comment:
I completely forgot about this function. That's of course very useful here :)

Reply:
That was being used already 😉

Reply:
Given that we use that, do we still need the tuple wrapping?

Reply:
> It seems useful to show that this is actually a `glib::Value`.

Why though? Again, `Rc` and `NonZero*` from the stdlib don't show it either, even `Cow` doesn't show that it's `Cow` (and whether it's borrowed or owned).

> Does this also print the type btw?

I have no idea. I guess I'll write a unit test.

Reply:
Well, first printing the type and then the value :)

Reply:
> Why though? Again, Rc and NonZero* from the stdlib don't show it either, even Cow doesn't show that it's Cow (and whether it's borrowed or owned).

Personally I prefer that so I know what I'm looking at when debugging things, but you have a point there.

@bilelmoussaoui @GuillaumeGomez What do you think?

Reply:
The whole point of a debug impl is to get information so I'd prefer what @sdroege is suggesting.

Reply:
The debug string only contains the value. Also I just realized that `debug_tuple("foo").field()` will quote the string again, which is not what we want here. With the current code,

```rust
let v = "foo".to_value();
assert_eq!(format!("{:?}", v), "foo");
```

fails with

```
thread 'value::tests::test_debug' panicked at 'assertion failed: `(left == right)`
  left: `"Value(\"\\\"foo\\\"\")"`,
 right: `"foo"`', glib/src/value.rs:1234:9
```

So now we know we can print both type (through `type_()`?) and value, does that mean I can go ahead with `{type}({debug_content})` and remove the `Value()` / `SendValue()`?

Reply:
Or maybe C syntax: `(type)value`? But OK with me then

Reply:
> The whole point of a debug impl is to get information

Yes, information that's not present in the code itself. You already know that you're debugging a `Value` because that's the type you declared yourself. But if you insist, I can print `Value()` around everything. In that case, how should the type be printed? Should it be `Value(<{type}> {debug_string})` / `Value({type}({debug_string}))` / something else?

Reply:
Is an extra space okay? I know that's not the usual formatting in C, but I find it really ugly otherwise. I've committed that for now, but if you want I'll remove it.

Reply:
I also thought of that but :man_shrugging: Whatever you prefer